manifests/repo.pp

```diff
@@ -35,7 +35,7 @@
           },
         }
 
-        Class['apt::update'] -> Package <| tag == 'syslog_ng' |>
```

Comment:
Could you please explain this change ?

Reply:
This is related to this failure https://github.com/ccin2p3/puppet-syslog_ng/runs/5365004194?check_suite_focus=true introduced by https://github.com/puppetlabs/puppetlabs-apt/commit/966ca80c56af071e53179d6d04cdd21f6523121f

```
        dependency cycles found: (Exec[apt_update] => Class[Apt::Update] => Package[apt-transport-https] => Class[Apt::Update] => Exec[apt_update])
```

1. the class `syslog_ng` include `syslog_ng::repo`
2. `syslog_ng::repo` declare a `apt::source`
3. `apt::source` use create_resource to declare a package `apt-transport-https` if it does not exist.

So if `apt-transport-https` is missing before the module is declared, the `Package['apt-transport-https']` resource will have tags `syslog_ng, syslog_ng::repo, apt::source` (scopes of the calling contexts).

Then the dependencies come in place:
```
Package['apt-transport-https'] -> Class['apt::update']
Class['apt::update'] -> Package <| tag == 'syslog_ng' |>
```

Because `Package['apt-transport-https']` is tagged `syslog_ng` this is a circular dependency.